### ISSUE TITLE
Suggestion: rename "CloudChannel" to "CloudQueue"?

### ISSUE THREAD

**Issue by dsyme** [mbraceproject/MBrace.Core#91]
On the whole I really like where the renaming and API reorganization has got us.

One thing I noticed today when coming back to 0.6.8 is about the CloudChannel name. As I mentioned in comments like [this](https://github.com/mbraceproject/MBrace.Core/pull/54#discussion_r27586417), I don't really love the "channel" terminology.   A channel in MBrace is ultimately a queue - why not call it a queue?   Won't people find that more intuitive? 

It seems to me that having a CloudQueue abstraction (e.g. which implements both ISendPort and IReceivePort and which has a name in the same way as CloudFile and CloudDicitonary) would make sense and be more regular?  The current use of send/receive pairs feels irregular.  And I get the feeling that performance considerations will ultimately require adding further operations to CloudQueue, e.g. supporting bulk operations.

**Comment by palladin** [comment]
If I understand correctly the suggestion is to have something like
```
let foo (q : CloudQueue) (msg : string) = cloud {
   do! CloudQueue.Send q msg
   let! msg = CloudQueue.Receive q
   return msg
}
```
it looks reasonable to me!

**Comment by palladin** [comment]
And maybe the name CloudChannel can cause more confusion about the semantics. CloudQueue has a much more simple explanation "it is just a queue plus details"!

**Comment by isaacabraham** [comment]
Agreed. At a glance I know what a CloudQueue is, more or less. The semantics are well established and having Send and Receive is easy to understand.

**Comment by eiriktsarpalis** [comment]
I suppose we then have to rename the methods to `Enqueue` and `Dequeue` as well?

**Comment by isaacabraham** [comment]
Not 100% essential IMHO but yes, that would be nice :-)

**Comment by dsyme** [comment]
> And maybe the name CloudChannel can cause more confusion about the semantics. CloudQueue has a much more simple explanation "it is just a queue plus details"!

Yes, that's it in a nutshell.  

> I suppose we then have to rename the methods to  Enqueue  and  Dequeue  as well?

That would make sense.

**Comment by eiriktsarpalis** [comment]
Fixed in ae47e4168d7d0cc001a25a36207ac6c99f38e1c6.

**Comment by palladin** [comment]
Great! I really like the new name!
